File: source/clibCore/CPAddonClass.vb

Option Explicit On
Option Strict On

Imports System.Runtime.InteropServices

Namespace Contensive.Core
    '
    ' comVisible to be activeScript compatible
    '
    <ComVisible(True)> _
    <ComClass(CPAddonClass.ClassId, CPAddonClass.InterfaceId, CPAddonClass.EventsId)> _
    Public Class CPAddonClass
        Inherits BaseClasses.CPAddonBaseClass
        Implements IDisposable
        '
#Region "COM GUIDs"
        Public Const ClassId As String = "6F43E5CA-6367-475C-AE65-FC988234922A"
        Public Const InterfaceId As String = "440D19E3-47A9-4CA2-B20C-077221015525"
        Public Const EventsId As String = "70B800AA-148A-4338-9EDB-70C85E1ADBDD"
#End Region
        '
        Private cp As CPClass
        Private Property UpgradeOK As Boolean
        '
        Friend Sub New(ByVal cp As CPClass)
            MyBase.New()
            Me.cp = cp
        End Sub
        '
        ' dispose
        '
        Protected Overridable Overloads Sub Dispose(ByVal disposing As Boolean)
            If Not Me.disposed Then
                Call appendDebugLog(".dispose, dereference main, csv")
                If disposing Then
                    '
                    ' call .dispose for managed objects
                    '
                    cp = Nothing
                End If
                '
                ' Add code here to release the unmanaged resource.
                '
            End If
            Me.disposed = True
        End Sub
        Protected disposed As Boolean = False
        '
        '
        '
        Public Overrides ReadOnly Property Admin() As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property ArgumentList() As String
            Get
                Return ""
            End Get
        End Property

        Public Overrides ReadOnly Property AsAjax() As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property BlockDefaultStyles() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property ccGuid() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property CollectionID() As Integer
            Get
                Return 0
            End Get
        End Property

        Public Overrides ReadOnly Property Content() As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property Copy() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property CopyText() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property CustomStyles() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property DefaultStyles() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property Description() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property DotNetClass() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property FormXML() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property Help() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property HelpLink() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property IconFilename() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property IconHeight() As Integer
            Get
                Return 0
            End Get
        End Property

        Public Overrides ReadOnly Property IconSprites() As Integer
            Get
                Return 0

            End Get
        End Property

        Public Overrides ReadOnly Property IconWidth() As Integer
            Get
                Return 0

            End Get
        End Property

        Public Overrides ReadOnly Property ID() As Integer
            Get
                Return 0

            End Get
        End Property

        Public Overrides ReadOnly Property InFrame() As Boolean
            Get

                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property IsInline() As Boolean
            Get
                Return False

            End Get
        End Property

        Public Overrides ReadOnly Property JavaScriptBodyEnd() As String
            Get

                Return ""
            End Get
        End Property

        Public Overrides ReadOnly Property JavascriptInHead() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property JavaScriptOnLoad() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property Link() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property MetaDescription() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property MetaKeywordList() As String
            Get

                Return ""
            End Get
        End Property

        Public Overrides ReadOnly Property Name() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property NavIconType() As String
            Get
                Return ""
            End Get
        End Property

        Public Overrides ReadOnly Property ObjectProgramID() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property OnBodyEnd() As Boolean
            Get
                Return False

            End Get
        End Property

        Public Overrides ReadOnly Property OnBodyStart() As Boolean
            Get
                Return False

            End Get
        End Property

        Public Overrides ReadOnly Property OnContentEnd() As Boolean
            Get
                Return False

            End Get
        End Property

        Public Overrides ReadOnly Property OnContentStart() As Boolean
            Get
                Return False

            End Get
        End Property

        Public Overrides Function Open(ByVal AddonId As Integer) As Boolean
            Return False

        End Function

        Public Overrides Function Open(ByVal AddonNameOrGuid As String) As Boolean

            Return False
        End Function

        Public Overrides ReadOnly Property OtherHeadTags() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property PageTitle() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property ProcessInterval() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property ProcessNextRun() As Date
            Get
                Return #12:00:00 AM#
            End Get
        End Property

        Public Overrides ReadOnly Property ProcessRunOnce() As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property RemoteAssetLink() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property RemoteMethod() As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property RobotsTxt() As String
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property ScriptCode() As String
            Get

                Return ""
            End Get
        End Property

        Public Overrides ReadOnly Property ScriptEntryPoint() As String 'Inherits BaseClasses.CPAddonBaseClass.ScriptEntryPoint
            Get

                Return ""
            End Get
        End Property

        Public Overrides ReadOnly Property ScriptLanguage() As String 'Inherits BaseClasses.CPAddonBaseClass.ScriptLanguage
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property SharedStyles() As String 'Inherits BaseClasses.CPAddonBaseClass.SharedStyles
            Get
                Return ""

            End Get
        End Property

        Public Overrides ReadOnly Property Template() As Boolean 'Inherits BaseClasses.CPAddonBaseClass.Template
            Get
                Return False
            End Get
        End Property
        '==========================================================================================
        ''' <summary>
        ''' Install an uploaded collection file from a private folder. Return true if successful, else the issue is in the returnUserError
        ''' </summary>
        ''' <param name="privateFolder"></param>
        ''' <param name="returnUserError"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overrides Function installCollectionFile(privateFolder As String, ByRef returnUserError As String) As Boolean
            Dim returnOk As Boolean = False
            Try
                returnOk = cp.core.addonInstall_installCollectionFile(privateFolder, returnUserError)
            Catch ex As Exception
                cp.core.handleExceptionAndRethrow(ex)
                If Not cp.core.siteProperties.trapErrors Then
                    Throw New ApplicationException("rethrow", ex)
                End If
            End Try
            Return returnOk
        End Function
        '
        Public Overrides Function installCollectionFromLibrary(collectionGuid As String, ByRef returnUserError As String) As Boolean
            Return False
        End Function
        '
        ' append to logfile
        '
        Private Sub appendDebugLog(ByVal copy As String)
            'My.Computer.FileSystem.WriteAllText("c:\clibCpDebug.log", Now & " - cp.addon, " & copy & vbCrLf, True)
            ' 'My.Computer.FileSystem.WriteAllText(System.AppDocmc.main_CurrentDocmc.main_BaseDirectory() & "cpLog.txt", Now & " - " & copy & vbCrLf, True)
        End Sub
        '
        ' testpoint
        '
        Private Sub tp(ByVal msg As String)
            'Call appendDebugLog(msg)
        End Sub
#Region " IDisposable Support "
        ' Do not change or add Overridable to these methods.
        ' Put cleanup code in Dispose(ByVal disposing As Boolean).
        Public Overloads Sub Dispose() Implements IDisposable.Dispose
            Dispose(True)
            GC.SuppressFinalize(Me)
        End Sub
        Protected Overrides Sub Finalize()
            Dispose(False)
            MyBase.Finalize()
        End Sub
#End Region
    End Class
End Namespace

File: source/clibCoreCs/Interfaces/cpApi/CPAddonClass.cs

using System;
using System.Reflection;
using System.Xml;
using System.Diagnostics;
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using Contensive.Core;
using Contensive.Core.Models.Entity;
using Contensive.Core.Controllers;
using static Contensive.Core.Controllers.genericController;
using static Contensive.Core.constants;
//
namespace Contensive.Core {
    public class CPAddonClass : BaseClasses.CPAddonBaseClass, IDisposable {
        //
        #region COM GUIDs
        public const string ClassId = "6F43E5CA-6367-475C-AE65-FC988234922A";
        public const string InterfaceId = "440D19E3-47A9-4CA2-B20C-077221015525";
        public const string EventsId = "70B800AA-148A-4338-9EDB-70C85E1ADBDD";
        #endregion
        //
        private CPClass cp;
        private bool UpgradeOK { get; set; }
        //
        public CPAddonClass(CPClass cp) : base() {
            this.cp = cp;
        }
        //
        // dispose
        //
        protected virtual void Dispose(bool disposing) {
            if (!this.disposed) {
                appendDebugLog(".dispose, dereference main, csv");
                if (disposing) {
                    //
                    // call .dispose for managed objects
                    //
                    cp = null;
                }
                //
                // Add code here to release the unmanaged resource.
                //
            }
            this.disposed = true;
        }
        protected bool disposed = false;
        //
        //====================================================================================================
        //
        public override bool Admin {
            get {
                return false;
            }
        }
        //
        //====================================================================================================
        //
        public override string ArgumentList {
            get {
                return "";
            }
        }
        //
        //====================================================================================================
        //
        public override bool AsAjax {
            get {
                return false;
            }
        }
        //
        //====================================================================================================
        //
        [Obsolete("This is no longer supported. Add a overriding style in another stylesheet instead of modifying", true)]
        public override string BlockDefaultStyles {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string ccGuid {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override int CollectionID {
            get {
                return 0;
            }
        }
        //
        //====================================================================================================
        //
        public override bool Content {
            get {
                return false;
            }
        }
        //
        //====================================================================================================
        //
        public override string Copy {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string CopyText {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string CustomStyles {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string DefaultStyles {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string Description {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string DotNetClass {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string FormXML {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string Help {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string HelpLink {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string IconFilename {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override int IconHeight {
            get {
                return 0;
            }
        }
        //
        //====================================================================================================
        //
        public override int IconSprites {
            get {
                return 0;

            }
        }
        //
        //====================================================================================================
        //
        public override int IconWidth {
            get {
                return 0;

            }
        }
        //
        //====================================================================================================
        //
        public override int ID {
            get {
                return 0;

            }
        }
        //
        //====================================================================================================
        //
        public override bool InFrame {
            get {

                return false;
            }
        }
        //
        //====================================================================================================
        //
        public override bool IsInline {
            get {
                return false;

            }
        }
        //
        //====================================================================================================
        //
        public override string JavaScriptBodyEnd {
            get {

                return "";
            }
        }
        //
        //====================================================================================================
        //
        public override string JavascriptInHead {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        [Obsolete("Create onready or onload events within your javascript. This method will be deprecated.", false)]
        public override string JavaScriptOnLoad {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string Link {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string MetaDescription {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string MetaKeywordList {
            get {

                return "";
            }
        }
        //
        //====================================================================================================
        //
        public override string Name {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string NavIconType {
            get {
                return "";
            }
        }
        //
        //====================================================================================================
        //
        public override string ObjectProgramID {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override bool OnBodyEnd {
            get {
                return false;

            }
        }
        //
        //====================================================================================================
        //
        public override bool OnBodyStart {
            get {
                return false;

            }
        }
        //
        //====================================================================================================
        //
        public override bool OnContentEnd {
            get {
                return false;

            }
        }
        //
        //====================================================================================================
        //
        public override bool OnContentStart {
            get {
                return false;

            }
        }
        //
        //====================================================================================================
        //
        public override bool Open(int AddonId) {
            return false;

        }
        //
        //====================================================================================================
        //
        public override bool Open(string AddonNameOrGuid) {

            return false;
        }
        //
        //====================================================================================================
        //
        public override string OtherHeadTags {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string PageTitle {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string ProcessInterval {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override DateTime ProcessNextRun {
            get {
                return Convert.ToDateTime("12:00:00 AM");
            }
        }
        //
        //====================================================================================================
        //
        public override bool ProcessRunOnce {
            get {
                return false;
            }
        }
        //
        //====================================================================================================
        //
        public override string RemoteAssetLink {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override bool RemoteMethod {
            get {
                return false;
            }
        }
        //
        //====================================================================================================
        //
        public override string RobotsTxt {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string ScriptCode {
            get {

                return "";
            }
        }
        //
        //====================================================================================================
        //
        public override string ScriptEntryPoint {
            get {

                return "";
            }
        }
        //
        //====================================================================================================
        //
        public override string ScriptLanguage {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override string SharedStyles {
            get {
                return "";

            }
        }
        //
        //====================================================================================================
        //
        public override bool Template {
            get {
                return false;
            }
        }
        //==========================================================================================
        /// <summary>
        /// Install an uploaded collection file from a private folder. Return true if successful, else the issue is in the returnUserError
        /// </summary>
        /// <param name="privatePathFilename"></param>
        /// <param name="returnUserError"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public override bool installCollectionFile(string privatePathFilename, ref string returnUserError) {
            bool returnOk = false;
            try {
                string ignoreReturnedCollectionGuid = "";
                returnOk = addonInstallClass.InstallCollectionsFromPrivateFile(cp.core, privatePathFilename, returnUserError, ignoreReturnedCollectionGuid, false, new List<string>());
            } catch (Exception ex) {
                cp.core.handleException(ex);
                if (!cp.core.siteProperties.trapErrors) {
                    throw;
                }
            }
            return returnOk;
        }
        //
        //====================================================================================================
        //
        public override bool installCollectionFromLibrary(string collectionGuid, ref string returnUserError) {
            return false;
        }
        //
        //====================================================================================================
        //
        private void appendDebugLog(string copy) {
            //
        }
        //
        //====================================================================================================
        //
        private void tp(string msg) {
            //
        }
        #region  IDisposable Support 
        // Do not change or add Overridable to these methods.
        // Put cleanup code in Dispose(ByVal disposing As Boolean).
        public void Dispose() {
            Dispose(true);
            GC.SuppressFinalize(this);
        }
        ~CPAddonClass() {
            Dispose(false);
            //INSTANT C# NOTE: The base class Finalize method is automatically called from the destructor:
            //base.Finalize();
        }
        #endregion
    }
}